infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java

```diff
@@ -179,6 +187,36 @@ KubernetesNamespace doCreateNamespaceAccess(String workspaceId, String name) {
         clientFactory, cheClientFactory, sharedPool.getExecutor(), name, workspaceId);
   }
 
+  @ScheduleRate(period = 1, initialDelay = 1, unit = TimeUnit.HOURS)
+  private void printLegacyWarningIfNeeded() {
+    if (!isNullOrEmpty(legacyNamespaceName)) {
+      LOG.warn(
+          "'che.infra.kubernetes.namespace' configuration parameter has been deprecated "
+              + "and is subject to remove in future releases.  Current value `{}`. Legacy workspaces located "
+              + "in this namespace may become unreachable in future releases. "
+              + "Please refer to the documentation about possible next steps.",
+          legacyNamespaceName);
+    }
+    if (LEGACY_NAMESPACE_NAME_PLACEHOLDERS.stream().anyMatch(defaultNamespaceName::contains)) {
+      LOG.warn(
+          "'che.infra.kubernetes.namespace.default' configuration parameter with `{}` "
+              + "placeholder has been deprecated and is subject to remove in  future releases."
+              + " Current value `{}`. Please refer to the documentation about "
+              + "possible next steps.",
+          LEGACY_NAMESPACE_NAME_PLACEHOLDERS,
+          defaultNamespaceName);
+    } else if (REQUIRED_NAMESPACE_NAME_PLACEHOLDERS
+        .stream()
+        .noneMatch(defaultNamespaceName::contains)) {
+      LOG.warn(
+          "Namespace strategies other than 'per user' has been deprecated and are subject to remove in  in future releases. "
+              + "Usage of one placeholder from `{}` is required in 'che.infra.kubernetes.namespace.default' parameter."
```

Comment:
Maybe something like "Using the <userid> or <username> placeholder is required ..."?